chalicelib/s3_connection.py

```diff
@@ -30,10 +33,37 @@ def get_object(self, key):
         # return found bucket content or None on an error
         try:
             response = self.client.get_object(Bucket=self.bucket, Key=key)
-            return response['Body'].read()
+            body = response['Body'].read()
+            return json.loads(body)
+        except json.JSONDecodeError:
+            return body
         except:
             return None
 
+    def get_size(self):
+        """
+        Gets the number of keys stored on this s3 connection. This is a very slow
+        operation since it has to enumerate all keys.
+        """
+        bucket = self.resource.Bucket(self.bucket)
+        return sum(1 for _ in bucket.objects.all())
+
+    def get_size_bytes(self):
+        """
+        Uses CloudWatch client to get the bucket size in bytes of this bucket.
+        Useful for checks - may need further configuration
+        """
+        now = datetime.datetime.utcnow()
+        resp = self.cw.get_metric_statistics(Namespace='AWS/S3',
+                                             MetricName='BucketSizeBytes',
+                                             Dimensions=[
+                                            {'Name': 'BucketName', 'Value': self.bucket},
+                                            {'Name': 'StorageType', 'Value': 'StandardStorage'}],
+                                            Statistics=['Average'],
+                                            Period=86400,
+                                            StartTime=(now-datetime.timedelta(days=1)).isoformat(),
```

Comment:
What is the significance of `StartTime` and `EndTime` here? Is it the average size over the past day as reported by Cloudwatch? Or the size of all objects created in the last day? Should probably add to docstring

Reply:
Explanation has been added to the docstring.